scripts/loadEmptyPrefsSets.js

```diff
@@ -0,0 +1,127 @@
+    fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    uuid = uuid || require("node-uuid");
+
+require("./shared/prefsSetsDbUtils.js");
+require("gpii-pouchdb");
+
+fluid.defaults("gpii.uuidLoader", {
+    gradeNames: ["gpii.pouch"],
+    outputFileName: process.env.OUTPUT_FILE || "generated-prefSets-" + new Date().toISOString(),
+    dbOptions: {
+        name: process.env.COUCHDB_URL
+    },
+    invokers: {
+        addPrefsSet: {
+            funcName: "gpii.uuidLoader.addPrefsSet",
+            args: [ "{that}", "{arguments}.0", "{arguments}.1"]
+        },
+        dbPut: {
+            funcName: "gpii.uuidLoader.dbPut",
+            args: [ "{that}", "{arguments}.0"]
+        }
+    }
+});
+
+gpii.uuidLoader.dbPut = function (that, doc) {
+    var promise = fluid.promise();
+    // We only perform the put after checking that the uuid doesn't exist in DB,
+    // just in case somebody wants to reuse this in the future.
+    //
+    that.get(doc._id).then(function (/* record */) {
+        promise.reject({isError: true, message: "doc with id", doc._id, "already exists in DB"});
+    }, function (err) {
+        if (err.message === "missing") {
+            that.put(doc).then(function (newDoc) {promise.resolve(newDoc);}, function (err) {promise.reject(err);});
+        } else {
+            promise.reject(err);
+        };
+    });
+    return promise;
+};
+
+gpii.uuidLoader.addPrefsSet = function (that, gpiiKey, prefSafe) {
+    var promise = fluid.promise();
+    var sequence = [
+        that.dbPut(gpiiKey),
+        that.dbPut(prefSafe)
+    ];
+
+    fluid.promise.sequence(sequence).then(function () {
+        promise.resolve(gpiiKey._id);
+    }, function (error) {
+        promise.reject(error);
+    });
+
+    return promise;
+};
+
+// The real action
+//
+function createNewUsers(uuidLoader) {
+    var data = gpii.prefsSetsDbUtils.generatePrefsSet(uuid.v4(), gpii.prefsSetsDbUtils.emptyPrefsSetBlock);
+    uuidLoader.addPrefsSet(data.key, data.prefsSafe).then(function (newUser) {
+        fs.appendFileSync(uuidLoader.options.outputFileName, newUser + "\n");
+        if (process.env.SHOW_PROGRESS) {
+            console.log("User",
+                        process.env.NEW_USERS - n + 1, "of",
+                        process.env.NEW_USERS, "created:",
+                        newUser);
+        }
+
+        if (n > 1) {
+            n--;
+            createNewUsers(uuidLoader);
+        } else {
+            console.log("We're done adding new users! :)");
+            console.log("Remember, you have a list of the created users here:", uuidLoader.options.outputFileName);
+        }
+    }, function (error) {
+        console.log("Got an error, let's stop it here. :/", error);
+        console.log("You can find the users we could create before crashing here:", uuidLoader.options.outputFileName);
+    });
+};
+
+// This triggers the "real action"
+//
+var n = parseInt(process.env.NEW_USERS);
```

Comment:
Namespace this variable too. All appearances of `process.env.NEW_USERS` at line 101, 102 can be replaced by this variable.

Reply:
Ah, i ses, you use `n` as a counter. I agree with @mrtyler, it's probably more readable to have 2 variables to track different things: 

```
gpii.loader.totalNumOfKeys = parseInt(process.env.NEW_USERS);
gpii.loader.count = 0;
```